library/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,623 @@
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            started=dict(
+                type='bool'
+            ),
+            purge=dict(
+                type='bool'
+            ),
+            purge_content_paths=dict(
+                type='list',
+                elements='str',
+                default=['/']
+            ),
+            profile_name=dict(
+                type='str',
+                required=True
+            ),
+            origin=dict(
+                type='dict',
+                options=origin_spec
+            ),
+            origin_host_header=dict(
+                type='str',
+            ),
+            origin_path=dict(
+                type='str',
+            ),
+            content_types_to_compress=dict(
+                type='list',
+                elements='str',
+            ),
+            is_compression_enabled=dict(
+                type='bool',
+                default=False
+            ),
+            is_http_allowed=dict(
+                type='bool',
+                default=True
+            ),
+            is_https_allowed=dict(
+                type='bool',
+                default=True
+            ),
+            query_string_caching_behavior=dict(
+                type='str',
+                choices=[
+                    'IgnoreQueryString',
+                    'BypassCaching',
+                    'UseQueryString',
+                    'NotSet'
+                ],
+                default='IgnoreQueryString'
+            ),
+        )
+
+        self.resource_group=None
+        self.name=None
+        self.state=None
+        self.started=None
+        self.purge=None
+        self.purge_content_paths=None
+        self.location=None
+        self.profile_name=None
+        self.origin=None
+        self.tags=None
+        self.origin_host_header=None
+        self.origin_path=None
+        self.content_types_to_compress=None
+        self.is_compression_enabled=None
+        self.is_http_allowed=None
+        self.is_https_allowed=None
+        self.query_string_caching_behavior=None
+
+        self.results = dict(changed=False)
+
+        super(AzureRMCdnendpoint, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                supports_check_mode=True,
+                                                supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        to_be_updated = False
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_cdnendpoint()
+
+        if self.state == 'present':
+
+            if not response and self.started is None:
```

Comment:
the if/elif seems little confusing. why not
if not response:
     if self.started is None:
     else:
elsle: